### PR TITLE
BaSP-TG-21: Time-sheets-tests

### PR DESCRIPTION
**Pull request by mara666**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "app_name is a system that provides Human Resources services to companies.",
   "main": "index.js",
   "scripts": {
-    "test": "jest",
+    "test": "jest --verbose",
```
Comment (lmalvasia) [inline package.json:7]: Is this necessary? Why did you add this?

File: src/tests/time-sheets.test.js [hunk 2]
```diff
@@ -1,5 +1,160 @@
...
+    test('should return status code 200', async () => {
+      const response = await request(app).get('/time-sheets/63535ae7fc13ae517a000031').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Time sheet found');
+    });
+  });
+
+  describe('GET BY ID ERROR /:id', () => {
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:41]: Move all the tests inside only one describe (we should have only one per method)

File: src/tests/time-sheets.test.js [hunk 3]
```diff
@@ -1,5 +1,160 @@
...
+    test('should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:42]: Why the request is returning a status 40?
You have to add the purpose why this is going to return an error.
```suggestion
    test('should return status code 400 when the id is invalid', async () => {
```

File: src/tests/time-sheets.test.js [hunk 4]
```diff
@@ -1,5 +1,160 @@
...
+      const response = await request(app).get('/time-sheets/634adaf83e995e6b4c7864z1').send();
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:47]: This equal to have expect(true).toBe(true). It's wrong.
You should have something like:
```suggestion
      expect(response.body.message).toEqual(`Invalid Id`);
```

File: src/tests/time-sheets.test.js [hunk 5]
```diff
@@ -1,5 +1,160 @@
...
+  describe('POST ERROR /time-sheets', () => {
+    const timeSheetMockedEmpty = {
+      description: '',
+      date: '',
+      hours: '',
+      task: '',
+      employee: '',
+      project: '',
+    };
+    test('should return status code 400', async () => {
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:78]: Why the request is returning a status 400?
You have to add the purpose why this is going to return an error.

File: src/tests/time-sheets.test.js [hunk 6]
```diff
@@ -1,5 +1,160 @@
...
+      const response = await request(app).post('/time-sheets').send(timeSheetMockedEmpty);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:83]: This is wrong.

File: src/tests/time-sheets.test.js [hunk 7]
```diff
@@ -1,5 +1,160 @@
...
+    });
+  });
+
+  describe('DELETE ERROR /:id', () => {
+    test('should return status code 400', async () => {
+      const response = await request(app).delete('/time-sheets/634adaf83e995e6b4c7864z1').send();
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
+      expect(response.body.data).toBeUndefined();
+      expect(response.body.message).toBe(`${response.body.message}`);
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:103]: This is wrong.

File: src/tests/time-sheets.test.js [hunk 8]
```diff
@@ -1,5 +1,160 @@
...
+    });
+  });
+
+  describe('DELETE /:id', () => {
+    test('should return status code 200', async () => {
+      const response = await request(app).delete('/time-sheets/634adbce3e995e6b4c7864ab').send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
```
Comment (leimeter-joaquin) [inline src/tests/time-sheets.test.js:91]: The error property is either true or false, probably best to use `.toBe(false)` I believe we want it to be false, not undefined or null or '' or 0, for example

File: src/tests/time-sheets.test.js [hunk 9]
```diff
@@ -1,5 +1,160 @@
...
+      date: '',
+      hours: '',
+      task: '',
+      employee: '',
+      project: '',
+    };
+    test('should return status code 400', async () => {
+      const response = await request(app).put('/time-sheets/63535ae7fc13ae517a000035').send(timeSheetEditEmpty);
+      expect(response.status).toBe(400);
+      expect(response.body.error).toBeTruthy();
```
Comment (leimeter-joaquin) [inline src/tests/time-sheets.test.js:155]: Same here as in my last comment

Comment (leimeter-joaquin) [reply src/tests/time-sheets.test.js:155]: Well, with true, of course

File: src/tests/time-sheets.test.js [hunk 10]
```diff
@@ -1,5 +1,160 @@
...
+      date: '2022-10-01',
+      hours: 10,
+      task: mongoose.Types.ObjectId('634adaf83e995e6b4c7864a7'),
+      employee: mongoose.Types.ObjectId('634d5803354e41cd60b9e400'),
+      project: mongoose.Types.ObjectId('6354c31fa0f546fd19325575'),
+    };
+    test('should return status code 200', async () => {
+      const response = await request(app).put('/time-sheets/63535ae7fc13ae517a000035').send(timeSheetMockedEdit);
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBeFalsy();
```
Comment (leimeter-joaquin) [inline src/tests/time-sheets.test.js:119]: Same here

**Review by leimeter-joaquin — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (mara666) [reply package.json:7]: I use this because in console doesn't show the describes!

File: src/tests/time-sheets.test.js [hunk 11]
```diff
@@ -1,5 +1,168 @@
...
+
+  describe('GET BY ID /time-sheets/:id', () => {
+    test('should return status code 200 if the time sheet _id supplied is found', async () => {
+      const existId = '63535ae7fc13ae517a000031';
+      const response = await request(app).get(`/time-sheets/${existId}`).send();
+      expect(response.status).toBe(200);
+      expect(response.body.error).toBe(false);
+      expect(response.body.data).toBeDefined();
+      expect(response.body.message).toBe('Time sheet found');
+    });
```
Comment (lmalvasia) [inline src/tests/time-sheets.test.js:39]: Add a new line between each test to separate them
For example
```
    test('should not create a project when there is not body', async () => {
       ....
    });

    test('should not create a project when the user send an invalid project name', async () => {
      ....
    });
```

**Review by leimeter-joaquin — APPROVED** [review]

**Review by kolodani — DISMISSED** [review]
awesome coworker

**Review by kolodani — APPROVED** [review]
all fix

**Pull request merged**